.travis.yml

```diff
@@ -8,6 +8,12 @@ services:
 python:
   - "2.7"
 
+env:
+  # run flake8 check on all python files in the project
+  - TEST_CMD="ci/flake8_diff.sh"
+  # run the full testsuite
+  - TEST_CMD="docker exec sardana-test sardanatestsuite"
+
```

Comment:
I just realised that this implementation makes the flake8 test also run a lot of the `install` part that is useless for it. This has 2 problems:
- It wastes time (the flake8 test could be much faster)
- It mixes cases for failure: e.g., if a commit introduces problems that make  `python setup.py install` to fail, the flake8 test would report an error that is actually due to the unrelated install issue

To fix this, we could use [stages](https://docs.travis-ci.com/user/build-stages) but maybe it is simpler to just do the install steps conditionally, as in the following example::

```yaml
(...)
env:
  - TEST_CMD="ci/flake8_diff.sh" INSTALL="N"
  - TEST_CMD="docker exec sardana-test sardanatestsuite"  INSTALL="Y"

(...)

install:
  - if [ $INSTALL == "Y" ]; then docker pull reszelaz/sardana-test ; fi
  - if [ $INSTALL == "Y" ]; then docker run -d --name=sardana-test -h sardana-test --volume=`pwd`:/sardana reszelaz/sardana-test ; fi
  - (...)

(...)


```